zebedee-cms/src/main/java/com/github/onsdigital/zebedee/model/publishing/PublishNotification.java

```diff
@@ -1,98 +1,202 @@
 package com.github.onsdigital.zebedee.model.publishing;
 
-import com.github.davidcarboni.httpino.Endpoint;
-import com.github.davidcarboni.httpino.Host;
-import com.github.davidcarboni.httpino.Http;
-import com.github.davidcarboni.httpino.Response;
+import com.github.davidcarboni.httpino.*;
 import com.github.onsdigital.zebedee.configuration.Configuration;
 import com.github.onsdigital.zebedee.exceptions.BadRequestException;
 import com.github.onsdigital.zebedee.json.ContentDetail;
 import com.github.onsdigital.zebedee.json.EventType;
 import com.github.onsdigital.zebedee.model.Collection;
+import com.github.onsdigital.zebedee.util.EncryptionUtils;
 import com.github.onsdigital.zebedee.util.SlackNotification;
+import com.github.onsdigital.zebedee.util.URIUtils;
 import com.github.onsdigital.zebedee.util.slack.PostMessageField;
+import com.google.gson.annotations.SerializedName;
+import org.apache.commons.lang.StringUtils;
+import org.apache.http.NameValuePair;
+import org.apache.http.message.BasicNameValuePair;
 import org.joda.time.DateTime;
 
 import java.text.SimpleDateFormat;
+import java.util.ArrayList;
 import java.util.Date;
 import java.util.List;
+import java.util.Arrays;
 
-import static com.github.onsdigital.logging.v2.event.SimpleEvent.info;
 import static com.github.onsdigital.logging.v2.event.SimpleEvent.error;
+import static com.github.onsdigital.logging.v2.event.SimpleEvent.info;
+import static com.github.onsdigital.zebedee.util.URIUtils.removeProtocolHostAndPort;
 
 /**
  * Created by bren on 16/12/15.
  * Publish notification to be sent to the website for caching timing purposes
  */
+
 public class PublishNotification {
```

Comment:
This class is now too big and convoluted, it has too many responsibilities. You should extract all the new code out in multiple classes. Everything that's related to the new Legacy Cache API should be in its own class. The code related to the Legacy Cache API in this file should be kept to a minimum. That will reduce coupling and improve readability.

Reply:
yes agree

Reply:
I moved the legacy cache api related code in a new class. Also created a new package to make it more obvious which class are new with the legacy cache api